### PR TITLE
Update DOD so that all PRs can select all the checks

### PR DESCRIPTION
**Pull request by raphaellueckl**
Opinionated PR:
============

In my opinion, in a DOD, you should select EVERYTHING before the PR is ready to merge. This was not the case before, because not all bullet points could be checked in all the cases. But this is true now. When you are ready to have your PR reviewed, you can first make sure that you checked the whole list and signed every point.

Some stuff was redundant (like the changelog things), one point was useless (because the pipeline makes sure that nobody can merge something not compatible with eslint rules or that breaks tests) and some others needed a bit more text so that they can be marked as done without lying.

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
+Fixes #(issue)
 
 # Done is when (DoD):
-- [ ] I have added UI/Unit tests for my changes
-- [ ] Old tests and eslint rule are not broken
-- [ ] I added modified component properties to the typescript typings
-- [ ] Design has been reviewed with (here name of reviewer)
+- [ ] My changes ship with tests, if needed.
+- [ ] I added modified component properties to the typescript typings, if needed.
+- [ ] Design changes have been reviewed with a designer.
```
Comment (01231) [inline PULL_REQUEST_TEMPLATE.md]: not always necessary ist it?

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
 - [ ] My changes are accordingly to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)
-- [ ] My changes are well documented in the README and showcased in the stories 
-- [ ] My changes are shortly added to CHANGELOG.md with the issue number
-- [ ] Tested with IE
+- [ ] My changes are well documented and showcased in the stories.
+- [ ] My changes are shortly added to CHANGELOG.md with the issue number.
+- [ ] Tested with IE, if affected.
```
Comment (01231) [inline PULL_REQUEST_TEMPLATE.md]: always needed in my opinon

Comment (raphaellueckl) [reply PULL_REQUEST_TEMPLATE.md]: Not if you make changes on the build level, or if you change readmes.

Comment (01231) [reply PULL_REQUEST_TEMPLATE.md]: True 👍

**Comment by 01231** [comment]
How about a section with criteria's which must be done for every single pr and a special section with criteria's which only apply to some PR's? Then we don't have to add a `if needed` to every sentence

**Review by danyball — DISMISSED** [review]

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,16 +1,11 @@
...
-- [ ] My changes are accordingly to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)
-- [ ] My changes are well documented in the README and showcased in the stories 
-- [ ] My changes are shortly added to CHANGELOG.md with the issue number
-- [ ] Tested with IE
-
-# Checklist:
-- [ ] My code follows the style guidelines of this project
-- [ ] I have performed a self-review of my own code
-- [ ] I have commented my code, particularly in hard-to-understand areas (focus on **WHY**)
+- [ ] The modifications are reflected locked tested.
```
Comment (LucaMele) [inline PULL_REQUEST_TEMPLATE.md]: @raphaellueckl what does this mean? `reflected locked tested.`

File: PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,16 +1,11 @@
...
+- [ ] A self-review of the code has been performed.
+- [ ] The modified component properties have been added to the typescript typings.
+- [ ] Potential design changes have been reviewed by a designer.
+- [ ] The modifications are well documented and showcased in the stories.
```
Comment (LucaMele) [inline PULL_REQUEST_TEMPLATE.md]: @raphaellueckl here i would still refer to the `README.md` so that is clear also for new devs

**Review by LucaMele — COMMENTED** [review]
I like @NoahLiechti Idea of splitting it, but also like this is fine for me.

I added 2 small comments and I personally would like to keep this:

`I have commented my code, particularly in hard-to-understand areas (focus on **WHY**)`

But no strong opinion

File: PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (raphaellueckl) [reply PULL_REQUEST_TEMPLATE.md]: My bad... thx

**Review by danyball — DISMISSED** [review]

File: PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (raphaellueckl) [reply PULL_REQUEST_TEMPLATE.md]: Changed to `Modifications within components are reflected by tests.`

**Review by danyball — APPROVED** [review]

**Review by 01231 — APPROVED** [review]

**Pull request merged**